File: 06-Encapsulation-And-Polymorphism/Shapes/Circle.cs
﻿using System;

public class Circle : IShape
    {
    private double radius;

    public Circle(double radius)
    {
        this.Radius = radius;
    }

    public double Radius
        {
        get { return radius; }
        set { radius = value; }
        }
    
    public double CalculateArea()
        {
        return Math.PI * this.Radius * this.Radius;
        }

    public double CalculatePerimeter()
    {
        return 2 * Math.PI*this.Radius;
    }
    }

File: 08-OOPInJava/Geometry/src/Circle.java
import java.util.ArrayList;

public class Circle extends PlaneShape {
    private double radius;

    public Circle(ArrayList<Vertex> vertices2D, double radius) {
        super(vertices2D);
        this.setVertex(vertices2D);
        this.setRadius(radius);
    }

    public void setVertex(ArrayList<Vertex> vertexArrayList) {
        if (vertexArrayList.size() != 1){
            throw new IllegalArgumentException("Circle needs only 1 vertex!");
        }
    }

    public void setRadius(double radius) {
        if (radius <= 0){
            throw new ArithmeticException("Radius can not be zero or negative!");
        }
        this.radius = radius;
    }

    @Override
    public double getArea() {
        return Math.PI * this.radius * this.radius;
    }

    @Override
    public double getPerimeter() {
        return 2 * Math.PI * this.radius;
    }
}
